src/main/java/games/strategy/engine/framework/startup/login/ClientLoginValidator.java

```diff
@@ -91,6 +115,26 @@ public String verifyConnection(final Map<String, String> propertiesSentToClient,
       return YOU_HAVE_BEEN_BANNED;
     }
     if (propertiesSentToClient.get(PASSWORD_REQUIRED_PROPERTY).equals(Boolean.TRUE.toString())) {
+      final String base64String = propertiesReadFromClient.get(ClientLogin.ENCRYPTED_PASSWORD_PROPERTY);
+      if (base64String != null) {
+        final String publicKey = propertiesSentToClient.get(RANDOM_RSA_PUBLIC_KEY_PROPERTY);
+        try {
+          final Cipher cipher = Cipher.getInstance(RSA_ECB_OAEPP);
+          cipher.init(Cipher.DECRYPT_MODE, rsaKeyMap.get(publicKey));
+          if (MessageDigest.isEqual(cipher.doFinal(Base64.getDecoder().decode(base64String)), m_password.getBytes())) {
```

Comment:
`MessageDigest#isEqual()`is a constant time algorithm, e.g. It always takes the same amount of time to compute depending on the length of both arrays no matter how much or little they differ.
This is to prevent timing attacks, but is no longer necessary when using the password as key to encrypt a random String.